src/server/HttpServerFactory.ts

```diff
@@ -4,5 +4,5 @@ import type { Server } from 'http';
  * A factory for HTTP servers
  */
 export interface HttpServerFactory {
-  startServer: (port: number) => Server;
+  startServer: (port: number | string) => Server;
```

Comment:
Are there still problems with doing this change? I see the `ServerInitializer` has a split between the two parameters so changing this interface should also be fine then right? Although it doesn't matter that much anymore, as in the v6.0.0 the architecture changes a bit so the port/socket is no longer passed to the server factory anyway. So if it's too much hassle it can stay like this for now.

Reply:
I can overload the method in the interface as suggested, but not in the implementation. 

![image](https://user-images.githubusercontent.com/53084151/201095445-b02619f2-37a9-4f50-b545-ee056cd8fb1d.png)

Reply:
After changing the interface, you want to do this in `BaseHttpServerFactory`:

```ts
  public startServer(port: number): Server;
  public startServer(socket: string): Server;
  public startServer(portOrSocket: number | string): Server {
   ....
```